engine/execution/ingestion/engine.go

```diff
@@ -325,9 +325,15 @@ func (e *Engine) reloadUnexecutedBlocks() error {
 
 		isRoot := rootBlock.ID() == last.ID()
 		if !isRoot {
-			err = e.reloadBlock(blockByCollection, executionQueues, lastExecutedID)
+			executed, err := state.IsBlockExecuted(e.unit.Ctx(), e.execState, lastExecutedID)
 			if err != nil {
-				return fmt.Errorf("could not reload the last executed final block: %v, %w", lastExecutedID, err)
+				return fmt.Errorf("cannot check is last exeucted final block has been executed %v: %w", lastExecutedID, err)
+			}
+			if !executed {
+				err = e.reloadBlock(blockByCollection, executionQueues, lastExecutedID)
```

Comment:
Could we add a `WARN` log here?

We've made the block execution transactional, so it should no happen that a block is partially executed. We could also add a comment to say, if this log never appear, we could consider removing the logic that load the last executed block.

Reply:
Good idea!